var/spack/repos/builtin.mock/packages/build-warnings/package.py

```diff
@@ -15,16 +17,33 @@ class BuildWarnings(Package):
     version("1.0", "0123456789abcdef0123456789abcdef")
 
     def install(self, spec, prefix):
-        with open("configure", "w") as f:
-            f.write(
-                """#!/bin/sh\n
-echo 'checking for gcc... /Users/gamblin2/src/spack/lib/spack/env/clang/clang'
-echo 'checking whether the C compiler works... yes'
-echo 'checking for C compiler default output file name... a.out'
-echo 'WARNING: ALL CAPITAL WARNING!'
-echo 'checking for suffix of executables...'
-echo 'foo.c:89: warning: some weird warning!'
-exit 1
-"""
-            )
-        configure()
+        if sys.platform == "win32":
+            with open("configure.bat", "w") as f:
+                f.write(
+                    """
```

Comment:
Do we need to turn echo off here? I'm surprised that the parsing works with the duplicated output.

Reply:
I think we probably should just to be as clear as possible. It has echo off now. The tests don't seem to mind either way though.

Reply:
Huh... well, either way thanks for making that change!